src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -482,9 +485,139 @@ EventEngine::TaskHandle PosixEventEngine::RunAfterInternal(
   return handle;
 }
 
+PosixEventEngine::PosixDNSResolver::PosixDNSResolver(
+    ResolverOptions const& options, PosixEnginePollerManager* poller_manager,
+    PosixEventEngine* event_engine)
+    : options_(options),
+      poller_manager_(poller_manager),
+      event_engine_(event_engine) {}
+
+PosixEventEngine::PosixDNSResolver::~PosixDNSResolver() {}
+
+using LookupTaskHandle = PosixEventEngine::PosixDNSResolver::LookupTaskHandle;
+
+LookupTaskHandle PosixEventEngine::PosixDNSResolver::LookupHostname(
+    LookupHostnameCallback on_resolve, absl::string_view name,
+    absl::string_view default_port, Duration timeout) {
+  GrpcAresHostnameRequest* request = new GrpcAresHostnameRequest(
```

Comment:
Rather than a raw `new` followed by an `Initialize`, consider creating a factory for `absl::StatusOr<GrpcAresHostnameRequest> Create(...)` creation, which could fail initialization.